frontend/src/components/module-viewer/ModuleViewer.tsx

```diff
@@ -155,12 +160,41 @@ const ModuleViewer = ({
   }, [courseID, moduleIndex]);
 
   useEffect(() => {
-    if (courseData) {
-      // Save to context
-      getLessons({
-        variables: { ids: courseData.course.modules[moduleIndex].lessons },
-      });
-    }
+    const fetchLessons = async () => {
+      const module = courseData?.course.modules[moduleIndex];
+      if (courseData) {
+        // Save to context
+        const { data: lessonRes } = await getLessons({
+          variables: { ids: module.lessons },
+        });
+        const { data: lessonProgressRes } = await getProgressData({
+          variables: {
+            userId: authenticatedUser?.id,
+            lessonIds: lessonRes.lessons.map(
+              (lesson: LessonResponse) => lesson?.id,
+            ),
+          },
+        });
+        const progresses = lessonProgressRes?.lessonProgress || [];
```

Comment:
For future-proof-ness, I'm wondering if you think it would be worth it to filter by non-null `completedAt` here. That way if one day we wanted to track when people start a lesson, this logic would still work.

Reply:
agree